server/events_test.go

```diff
@@ -1144,3 +1188,20 @@ func TestServerEventsPingStatsZ(t *testing.T) {
 		t.Fatalf("Error unmarshalling the statz json: %v", err)
 	}
 }
+
+func TestGatewayNameClientInfo(t *testing.T) {
+	sa, _, sb, _, _ := runTrustedCluster(t)
+	defer sa.Shutdown()
+	defer sb.Shutdown()
+
+	_, _, l := newClientForServer(sa)
+
+	var info Info
+	err := json.Unmarshal([]byte(l[5:]), &info)
+	if err != nil {
+		t.Fatalf("Could not parse INFO json: %v\n", err)
+	}
+	if info.Cluster != "TEST CLUSTER 22" {
```

Comment:
I understand from above that info.Cluster would be set if gateway is enabled, which I am not sure how runTrustedCluster() is doing that. Yet test is passing, so I guess I am missing something.

Reply:
Line 101.